File: Hot100/字母异位词分组.cs
 public IList<IList<string>> GroupAnagrams(string[] strs) {
        List<IList<string>> res = new List<IList<string>>();
        Dictionary<string,List<string>> dic_word= new Dictionary<string,List<string>>();
        foreach(string str in strs){
            char [] chars= str.ToCharArray();
            Array.Sort(chars);
            string sortword = new  string(chars);
            if(dic_word.ContainsKey(sortword)){

                dic_word[sortword].Add(str);
            }
            else{
                List<string> temp = new List<string>(){ str};
                dic_word.Add(sortword,temp);
            }
        }
        foreach( List<string> value in dic_word.Values){
            res.Add(value);
        }
        return res;
    }

File: Hot100/字母异位词分组.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let res = [];
    let myMap = new Map();
    for(let word of strs){        
        let sortword = sortstring(word);
        if(myMap.has(sortword)){
            myMap.get(sortword).push(word);
            console.log(myMap.get(sortword))
        }
        else{
            myMap.set(sortword,[word]);
        }
    }
   for (let value of myMap.values()) {
        res.push(value);
    }
    return res;
};

//对字符串排序
function sortstring(str){
    let split_Str = str.split('');
    split_Str.sort();
    return split_Str.join('');
}